### PR TITLE
Control Panel and Apps Platform Documentation

### PR DESCRIPTION
**Pull request by Bani2791**
# 1. App Store

**Setup :material-menu-right: App Store**

The **ConnexCS App Store** lets you discover, download, and install applications on your user account.

The Apps created using the IDE when published and are available for use in the App Store.

The Apps can be created by the other customers or developers and can be used by ConnexCS's customers.

## Steps to install Apps

1. Login to your account. Navigate to Setup :material-menu-right: App Store. Click on the App you wish to install.

2. Click on **Install**.

3.After hitting the Install button. A window appears which has 2 fields, **App Name** you can either change the name of the application or keep it same. The second allows you to select the  **Version** of the application.

[4.Click](http://4.click/) on **Install** and the App will be installed on your user account.

[5.You](http://5.you/) can view the **Description**, **Installed Versions**, **All Versions** and any **Documentation** related to the Application.

[App Store - ConnexCS Documentation (bani-controlpanel--connexcs-docs.netlify.app)](https://bani-controlpanel--connexcs-docs.netlify.app/setup/appstore/)

# 2. Performance based CLI
5. **Performance Top Batch Size** choose how many of the top CLIs we use together.

6. **Performance Interval**: Choose from the dropdown how long this batch stays active(in seconds). You can choose from the minimum value of 5 minutes and maximum value of 1 day with intervals of 300 seconds.

7. **Performance Ban Time**: How long a used DID is paused for. You can choose the value from the dropdown with minimum value being 5 minutes and maximumm value being 90 days. The value is stored in **seconds**.

[CLI - ConnexCS Documentation (bani-controlpanel--connexcs-docs.netlify.app)](https://bani-controlpanel--connexcs-docs.netlify.app/customer/cli/#performance-based-cli-selection)

# 3. Logging

### Decode Identity header in SIP Trace

1. Click on the **INVITE** packet in the SIP Trace.
2. A window will appear with **Raw** and **Shaken** tabs. Only if you are signing calls using **Shaken** certificate.
3. The **Shaken** tab will have the decoded information from the **Raw** tab like **Algorithm**, **Public Certificate URL**, **Attestation Level**, **Dialled Number**, **CLI**, **Unique Customer ID**, **Timestamp**.

[Logging - ConnexCS Documentation (bani-controlpanel--connexcs-docs.netlify.app)](https://bani-controlpanel--connexcs-docs.netlify.app/logging/#decode-identity-header-in-sip-trace)

# 4. Anyedge>Backup Zones and Limit Zones
* **Destination**: Enter the Customer's server's IP address.
* **Limit Zones**: The calls hitting the selected zones in the Limit Zones field will go directly to the entered IP address(server) in the Destination field.
* **Backup Zones**: Whenever the server of the main zone fails, the traffic will route to the zone selected in the Backup Zones field.

[AnyEdge - ConnexCS Documentation (bani-controlpanel--connexcs-docs.netlify.app)](https://bani-controlpanel--connexcs-docs.netlify.app/anyedge/anyedge/#anyedge-destinations)

# 5. Circuit Test
|Parameter|Description|
|---------|-----------|
|**Created**| When the log was created mm/dd/yyyy|
|**Provider**|List of your providers|
|**MOS- Mean Opinion Score**|To find the call quality|
|**DTMF- Dual Tone Multi Frequency**|diagnose and resolve various communication problems, enhancing network performance and ensuring smooth communication experiences|
|**Answered**|Call has been connected and answered|
|**Destroyed**|Call has been killed|
|**Direction**|Either **Termination** or **Origination**|
|**Card**|Displays Card-ID/Name|
|**Destination Number**|Recipient's number |
|**Caller**|1. **ID**: Caller's ID number <br> 2. **Display Name** of the caller <br> 3. **IP**: IP address of the caller|
|**Called**|1. **ID**: Recipient's ID number <br> 2. **Display Name** of the recipient <br> 3. **IP**: IP address of the recipient|
|**SIP Call ID**|SIP call ID number|
|**Disconnection**|1. **Code**: standardized code used to identify the reason why a call was disconnected. <br> 2. **Status**: Specific reason why a call was terminated unexpectedly. <br> These **codes** and **status **are crucial for troubleshooting issues, analyzing network performance, and understanding call quality.|
|**Released By**|BYE message sent by- Upstream or Downstream |
|**Answer Delay**|Time it takes between the initiation of a call and the moment the called party answers. It's a crucial metric used to gauge responsiveness and efficiency in call handling, impacting both user experience and operational effectiveness.|
|**_100_Delay**|Delay between an INVITE message and 100 Trying request|
|**_180_Delay**|Delay between 100 Trying and 180 Ringing request|
|**_183_Delay**|Delay between 180 Ringing and 183 Session Progress|
|**_100_183_Delay**|Delay between 100 Trying and 183 Session Progress|
|**Audio Signal Delay (Latency)**| Time it takes for an audio signal to travel from the sender to the receiver. Its a crucial factor in the quality of voice and video communication in telecommunication systems|
|**Latest RBT Delay**|The delay between selecting a Ring Back Tone (RBT) and it actually playing for callers can vary significantly|
|**Early Media Peak Signal Level**|The highest volume level measured during the initial audio exchange before the call is fully connected. This information could be useful for QoS, Network troubleshooting, Media optimization|
|**Active Media Peak Signal Level**|Refers to the highest power level attained by the audio or video signal during a call or streaming session| This information could be useful for Audio/video quality, Network performance, Regulation compliance|
|**Recorded**| Recorded received call file name, transmitted call file name, mix file name|
|**Answered Duration**| Duration of the answered call|
|**Current Calls Count**|Number of current calls|
|**SIP Pcap FileName**|Filename of a SIP Pcap (packet capture) file|
|**SIP and RTP Pcap FileName**|Filename of a SIP and RTP Pcap (packet capture) file|
|**RTP Codec**|Name of the RTP Codec used in the call|
|**RTP Caller**|1. **Address** of the caller. <br> 2. **Port** used by the caller for call sessions. <br> 3. **Synchronization Source (SSRC)** identifier for a specific source of synchronized multimedia data within an RTP session <br> 4. **G107R** is a codec <br> 5. **G107MOS** is a codec <br> 6. **Packets Count**: Number of packets. <br> 7. **Lost Packets**: Number of packets lost in transmission <br> 8. **Max RFC3550 Jitter**: Refers to the maximum value of the calculated jitter for an RTP stream over a specified measurement interval. High jitter can negatively impact the quality of real-time applications such as voice and video calls, as it can lead to stuttering, delays, and distortions. <br> 9. **Mean RFC3550 Jitter**: Refers to the average deviation of the time it takes for consecutive RTP packets to arrive at the receiver, compared to the sender's timestamps. It's a measure of the variability in the packet arrival times, which can be caused by various factors such as network congestion, routing changes, and packet loss. <br> 10. **Max Delta**: refers to the largest difference in arrival times between two consecutive RTP packets received at the receiver. It essentially measures the maximum variation in the time it takes for packets to arrive, which can be an indicator of network issues like congestion or variable latency. <br> 11. **Min Delta**: Minimum time difference between RTP packets. <br> 12. **Packet Time**: The time packet was sent or received <br> 13. **Non Silent Percentage**: Proportion of time during a call when there is actual conversation happening <br> 14. **Delay**: Time it takes for an RTP packet to travel from the sender to the receiver|
|**RTP Called**|1. **Address** of the recipient. <br> 2. **Port** used by the recipient for call sessions. <br> 3. **Synchronization Source (SSRC)** identifier for a specific source of synchronized multimedia data within an RTP session <br> 4. **G107R** is a codec <br> 5. **G107MOS** is a codec <br> 6. **Packets Count**: Number of packets. <br> 7. **Lost Packets**: Number of packets lost in transmission <br> 8. **Max RFC3550 Jitter**: Refers to the maximum value of the calculated jitter for an RTP stream over a specified measurement interval. High jitter can negatively impact the quality of real-time applications such as voice and video calls, as it can lead to stuttering, delays, and distortions. <br> 9. **Mean RFC3550 Jitter**: Refers to the average deviation of the time it takes for consecutive RTP packets to arrive at the receiver, compared to the sender's timestamps. It's a measure of the variability in the packet arrival times, which can be caused by various factors such as network congestion, routing changes, and packet loss. <br> 10. **Max Delta**: refers to the largest difference in arrival times between two consecutive RTP packets received at the receiver. It essentially measures the maximum variation in the time it takes for packets to arrive, which can be an indicator of network issues like congestion or variable latency. <br> 11. **Min Delta**: Minimum time difference between RTP packets. <br> 12. **Packet Time**: The time packet was sent or received <br> 13. **Non Silent Percentage**: Proportion of time during a call when there is actual conversation happening <br> 14. **Delay**: Time it takes for an RTP packet to travel from the sender to the receiver|
|**RTCP RTT**|1. **Max**: This represents the longest measured round-trip time for any of the received RTCP packets during a specific interval. <br> 2. **Min**: This represents the shortest measured round-trip time for any of the received RTCP packets during a specific interval.|
|**RTCP Caller**|1. **Jitter Max**: Highest recorded difference between the arrival times of consecutive RTP packets received from the caller (sender). It signifies the maximum variation in packet arrival times, which can contribute to audio/video distortion and choppiness. <br> 2. **Jitter Mean**: This value represents the average difference between the arrival times of consecutive RTP packets received from the caller. It provides a general sense of the overall variability in packet arrival times, impacting perceived quality. <br> 3. **Lost Packets**: This value indicates the number of RTP packets sent by the caller that weren't received by the receiver.|
|**RTCP Called**|1. **Jitter Max**: largest observed absolute difference between consecutive packet arrival times, expressed in the same units as the sender's timestamp clock. <br> 2. **Jitter Mean**: Average amount of jitter experienced by the receiver, calculated as the average of the absolute differences between consecutive packet arrival times, again using the sender's timestamp units. <br> 3. **Lost Packets**: Total number of packets that the receiver didn't receive out of the expected number sent by the sender.|
|**Test ID**|Randomly generated ID for the created test|
|**RTP Rx Packets**|1. **Total**: Total number of received packets. <br> 2. **Discarded**: Total number of discarded received packets. <br> 3. **Discarded Duplicated**: Total number of duplicated discarded received packets. <br> 4. **Discarded Early**: Refers to RTP packets that are received but not processed or delivered by the receiving party. These packets are dropped early on, before they even reach the application handling the call (like a video conferencing app). <br> 5. **Discarded Late**: Refers to packets that are dropped or ignored because they arrived after a certain threshold of time has passed. <br> 6. **Discarded Error**: Indicates that a packet with a detected error was not processed by the receiver. This can happen due to the severity of the error, limited resources, or specific configuration settings.|
|**RTP Codec Sample Rate**| Refers to the number of times per second that an audio or video signal is sampled for encoding and decoding within the RTP framework. It plays a crucial role in determining the quality and bandwidth requirements of the media content being transmitted.|
|**Ingress ID**|Ingress ID of the call|
|**RTCP caller DSCP**|RTP caller DSCP refers to the Differentiated Services Code Point (DSCP) value assigned to outgoing RTP media packets by the sender (caller). DSCP is a mechanism used in network traffic management to prioritize specific types of traffic based on predefined classes.|
|**RTCP called DSCP**|RTP called DSCP refers to the DSCP value assigned to incoming RTP media packets to the recipient|

[Circuit Test - ConnexCS Documentation (bani-controlpanel--connexcs-docs.netlify.app)](https://bani-controlpanel--connexcs-docs.netlify.app/circuit-test/#view-modification)

# 6. Customer > Routing>CLI
##### Performance based CLI Selection

The Performance based CLI Selection chooses the CLI with the best ASR from a large list or a database.

This feature's objective is to allow customers to utilise the best performing CLIs.

**How does this algorithm work?**

1. If there are more than 1000 records in the CLI Database, the system will randomly pick 1000 DIDs, then pick the best ones from that list of best performing ones.
2. An ASR lookup is initiated and the ASR reading requires at least 50 calls (per CLI). It means the system will then shortlist 50 calls (out of 1000) with the best ASR.
3. The system uses a **Sliding Window** concept over 7 days to achieve the ASR readings. This concept will add up the all the calls (successful and unsuccessful) per CLI to give us the ASR readings. Once the system gets all the CLIs ASR readings, it will arrange it from Best to Worse and will pick the top 5 CLIs.
4. Out of the 1000 calls the system will pick the top 5 performing CLIs every 5 minutes.
5. If there are less than 50 calls to this CLI (or none), it will assume an ASR of 50%.

[CLI - ConnexCS Documentation (bani-controlpanel--connexcs-docs.netlify.app)](https://bani-controlpanel--connexcs-docs.netlify.app/customer/cli/#performance-based-cli-selection)

# 7. Recalculate Call Detail Record
* **Min Duration** (for Adjust Call Duration operations) (Calls with Minimum Duration in the CDR for example 3600 seconds will be considered for re-calculation)

* **New Duration** (for Adjust Call Duration operations) (All the calls with minimum duration of 3600 seconds will be recalculated with the value in the New Duration, for example 60 seconds).
(All the calls will be converted from 3600 seconds to 60 seconds and the billing will also be redone according to 60 seconds duration).

!!! Example
    |Duration(seconds)|Cost per second($)|Total Cost($)|
    |-----------------|---------------|----------------|
    |3600(minimum duration)|0.0001|0.36|
    |60(new duration)|0.0001|0.006|

[CDR - ConnexCS Documentation (bani-controlpanel--connexcs-docs.netlify.app)](https://bani-controlpanel--connexcs-docs.netlify.app/customer/cdr/)

# 8. Apps Platform>Query Builder>G3 query builder

### G3 Query Builder

If you wish to generate a graph, for example from past 10 days, you can use the feature `start_date`.

`start_date` is a natural language date parser in Javascript. It can extract data from any provided text and handle the majority of date/time formats.

1. Today, Tomorrow, Yesterday, Last Friday, etc
2. 1 January 2013 - 9 May August 2013
3. This Tuesday from 3:00 - 6.00
4. 7 days ago
5. 4 weeks from now
6. Sun Aug 7 2013 18:40:39 UTC/GMT +5:30 hours (IST)
7. 2024-11-30T08:15:30-05:30

|Normal SQL Query|G3 Query|
|----------------|--------|
|`from `cdr`.`cdr` where `dt` >= DATE_SUB(NOW(), INTERVAL 30 DAY)`|start_date: 7 days ago|

You can also put a default value of the `start_date` by making it a variable using **Set Vars**.

For example,

```js
start_date: 7 days ago // G3 query
```

```js
start_date: :start_date // making it a variable using Set Vars
```

You can put any value in the Vars field, it will be treated as a default value when you use the above code.

[Query Builder - ConnexCS Documentation (bani-controlpanel--connexcs-docs.netlify.app)](https://bani-controlpanel--connexcs-docs.netlify.app/apps/architecture/query-builder/#g3-query-builder)

# 9. Webphone > Environmental Variables

Here, you have the capability to set the [Environmental Variables](https://docs.connexcs.com/apps/architecture/environmental-variables/) which can be called as a **function** (internal or external) from the **Dynamic Data** option in the **Component Attribute** from the **Page Builder**.

!!! Example
    1. Configure the **Environmental Variables** as explained [here](https://docs.connexcs.com/apps/architecture/environmental-variables/#steps-for-configuring-the-variables).
    2. Navigate to [Page Builder](https://docs.connexcs.com/apps/page-builder/), select the component, click on [Component Attribute](https://docs.connexcs.com/apps/page-builder/#component-attribute), then **Option :material-menu-right: Dynamic Data :material-menu-right: Function**, click on Function (can be internal or external). You can call the environmental variables using `this.env`. You can create the function in [Form Attribute :material-menu-right: Action Panel](https://docs.connexcs.com/apps/page-builder/#form-attribute).

[Webphone - ConnexCS Documentation (bani-controlpanel--connexcs-docs.netlify.app)](https://bani-controlpanel--connexcs-docs.netlify.app/apps/components/webphone/)

### CODE REVIEW

File: docs/setup/appstore.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# App Store
+
+**Setup :material-menu-right: App Store**
+
+The **ConnexCS App Store** lets you discover, download, and install applications on your user account.
+
+The Apps created using the IDE when published and are available for use in the App Store.
```
Comment (AdamNawaz-connex) [inline docs/setup/appstore.md:7]: The Apps created using the IDE are available for use in the App Store when published.

File: docs/setup/appstore.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+The Apps can be created by the other customers or developers and can be used by ConnexCS's customers.
```
Comment (AdamNawaz-connex) [inline docs/setup/appstore.md:9]: The Apps can be created by other customers or developers and used by ConnexCS's customers.

File: docs/setup/appstore.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+## Steps to install Apps
+
+1. Login to your account. Navigate to Setup :material-menu-right: App Store. Click on the App you wish to install.
+2. Click on **Install**.
+
+<img src= "/setup/img/appstore.png">
+
+3.After hitting the Install button. A window appears which has 2 fields, **App Name** you can either change the name of the application or keep it same. The second allows you to select the  **Version** of the application.
```
Comment (AdamNawaz-connex) [inline docs/setup/appstore.md:18]: After hitting the 'Install' button, a window appears with two fields. The first field is 'App Name', where you can either change the name of the application or keep it the same. The second field allows you to select the version of the application.

File: docs/anyedge/anyedge.md [hunk 4]
```diff
@@ -69,6 +69,12 @@ It can provide added **Transport Layer Security (TLS)/ Secure Sockets Layer (SSL
 
 Click :material-plus: button to specify the Destination IP, and one or more Limit (Primary) and Backup (Secondary) Zones.
 
+* **Destination**: Enter the Customer's server's IP address.
+* **Limit Zones**: The calls hitting the selected zones in the Limit Zones field will go directly to the entered IP address(server) in the Destination field.
```
Comment (AdamNawaz-connex) [inline docs/anyedge/anyedge.md:73]: add something like this Limit Zones control access to specific servers within a load balancer. By defining limit zones, you restrict access to certain servers from designated zones. For example, if Sydney is not added to the limit zone configuration, individuals from Sydney will be unable to access this designated server within the load balancer.

File: docs/anyedge/anyedge.md [hunk 5]
```diff
@@ -69,6 +69,12 @@ It can provide added **Transport Layer Security (TLS)/ Secure Sockets Layer (SSL
...
+* **Backup Zones**: Whenever the server of the main zone fails, the traffic will route to the zone selected in the Backup Zones field.
+
+<img src= "/anyedge/anyedge5.png">
```
Comment (AdamNawaz-connex) [inline docs/anyedge/anyedge.md:76]: Blur the IP address

File: docs/apps/architecture/query-builder.md [hunk 6]
```diff
@@ -34,3 +34,39 @@ Common use cases include: Database Administration Tools, Web Application, Report
 8.Click on `Save` and then `Run`. It will display a query output.
 
 <img src= "/apps/img/query5.png">
+
+### G3 Query Builder
+
+If you wish to generate a graph, for example from past 10 days, you can use the feature `start_date`.
```
Comment (AdamNawaz-connex) [inline docs/apps/architecture/query-builder.md:40]: When creating a query, you can select the data source as G3, which is helpful when you intend to visualize the data as a graph. The start_date feature facilitates specifying time ranges using natural language parsing in JavaScript.

File: docs/circuit-test.md [hunk 7]
```diff
@@ -44,6 +44,49 @@ View a history of completed circuit tests, including the provider the call route
 + **Filters** pop out on the right allowing you to filter some of your data.
 + Adjust the Column ordering.
 
+|Parameter|Description|
+|---------|-----------|
+|**Created**| When the log was created mm/dd/yyyy|
+|**Provider**|List of your providers|
+|**MOS- Mean Opinion Score**|To find the call quality|
```
Comment (AdamNawaz-connex) [inline docs/circuit-test.md:51]: Used to evaluate the perceived quality of audio.

File: docs/circuit-test.md [hunk 8]
```diff
@@ -44,6 +44,49 @@ View a history of completed circuit tests, including the provider the call route
...
+|**DTMF- Dual Tone Multi Frequency**|diagnose and resolve various communication problems, enhancing network performance and ensuring smooth communication experiences|
```
Comment (AdamNawaz-connex) [inline docs/circuit-test.md:52]: DTMF is detected or Not Detected in this test

File: docs/circuit-test.md [hunk 9]
```diff
@@ -44,6 +44,49 @@ View a history of completed circuit tests, including the provider the call route
...
+|**Answered**|Call has been connected and answered|
```
Comment (AdamNawaz-connex) [inline docs/circuit-test.md:53]: Indicates that the call has been successfully connected and answered.

File: docs/circuit-test.md [hunk 10]
```diff
@@ -44,6 +44,49 @@ View a history of completed circuit tests, including the provider the call route
...
+|**Destroyed**|Call has been killed|
+|**Direction**|Either **Termination** or **Origination**|
```
Comment (AdamNawaz-connex) [inline docs/circuit-test.md:55]: Specifies whether the call is Termination or Origination.

File: docs/circuit-test.md [hunk 11]
```diff
@@ -44,6 +44,49 @@ View a history of completed circuit tests, including the provider the call route
...
+|**Card**|Displays Card-ID/Name|
+|**Destination Number**|Recipient's number |
```
Comment (AdamNawaz-connex) [inline docs/circuit-test.md:57]: The recipient's phone number.

File: docs/circuit-test.md [hunk 12]
```diff
@@ -44,6 +44,49 @@ View a history of completed circuit tests, including the provider the call route
...
+|**Caller**|1. **ID**: Caller's ID number <br> 2. **Display Name** of the caller <br> 3. **IP**: IP address of the caller|
+|**Called**|1. **ID**: Recipient's ID number <br> 2. **Display Name** of the recipient <br> 3. **IP**: IP address of the recipient|
+|**SIP Call ID**|SIP call ID number|
+|**Disconnection**|1. **Code**: standardized code used to identify the reason why a call was disconnected. <br> 2. **Status**: Specific reason why a call was terminated unexpectedly. <br> These **codes** and **status **are crucial for troubleshooting issues, analyzing network performance, and understanding call quality.|
```
Comment (AdamNawaz-connex) [inline docs/circuit-test.md:61]: Standardized SIP response codes

File: docs/circuit-test.md [hunk 13]
```diff
@@ -44,6 +44,49 @@ View a history of completed circuit tests, including the provider the call route
...
+|**Released By**|BYE message sent by- Upstream or Downstream |
```
Comment (AdamNawaz-connex) [inline docs/circuit-test.md:62]: Indicates whether the call termination was initiated by Upstream or Downstream.

File: docs/circuit-test.md [hunk 14]
```diff
@@ -44,6 +44,49 @@ View a history of completed circuit tests, including the provider the call route
...
... an RTP stream over a specified measurement interval. High jitter can negatively impact the quality of real-time applications such as voice and video calls, as it can lead to stuttering, delays, and distortions. <br> 9. **Mean RFC3550 Jitter**: Refers to the average deviation of the time it takes for consecutive RTP packets to arrive at the receiver, compared to the sender's timestamps. It's a measure of the variability in the packet arrival times, which can be caused by various factors such as network congestion, routing changes, and packet loss. <br> 10. **Max Delta**: refers to the largest difference in arrival times between two consecutive RTP packets received at the receiver. It essentially measures the maximum variation in the time it takes for packets to arrive, which can be an indicator of network issues like congestion or variable latency. <br> 11. **Min Delta**: Minimum time difference between RTP packets. <br> 12. **Packet Time**: The time packet was sent or received <br> 13. **Non Silent Percentage**: Proportion of time during a call when there is actual conversation happening <br> 14. **Delay**: Time it takes for an RTP packet to travel from the sender to the receiver|
```
Comment (AdamNawaz-connex) [inline docs/circuit-test.md:78]: Max Delta: refers    Refers   ( R is capital)

File: docs/customer/cdr.md [hunk 15]
```diff
@@ -33,10 +32,18 @@ When viewing CDRs for a specific customer, use the **`Recalc CDR`** button to re
 
 * **Release Reason** (for Rerate operations)- Select the reason for the call's termination (multiple selections allowed). This will revise the amount charged for the calls.
 
-* **Min Duration** (for Adjust Call Duration operations)
+* **Min Duration** (for Adjust Call Duration operations) (Calls with Minimum Duration in the CDR for example 3600 seconds will be considered for re-calculation)
```
Comment (AdamNawaz-connex) [inline docs/customer/cdr.md:35]: Minimum Duration of calls that will be considered for re-calculation

File: docs/customer/cdr.md [hunk 16]
```diff
@@ -33,10 +32,18 @@ When viewing CDRs for a specific customer, use the **`Recalc CDR`** button to re
...
+
+* **New Duration** (for Adjust Call Duration operations) (All the calls with minimum duration of 3600 seconds will be recalculated with the value in the New Duration, for example 60 seconds).
```
Comment (AdamNawaz-connex) [inline docs/customer/cdr.md:37]: The duration of qualifying calls will be recalculated to this value.

File: docs/customer/cdr.md [hunk 17]
```diff
@@ -33,10 +32,18 @@ When viewing CDRs for a specific customer, use the **`Recalc CDR`** button to re
...
+(All the calls will be converted from 3600 seconds to 60 seconds and the billing will also be redone according to 60 seconds duration).
```
Comment (AdamNawaz-connex) [inline docs/customer/cdr.md:38]: No need for this

File: docs/customer/cli.md [hunk 18]
```diff
@@ -268,26 +268,32 @@ Our Userspace database allows you to manage large lists of numbers. Once you upl
...
+
 **How does this algorithm work?**
 
-1. If there is more than 1000 records in the CLI Database, it will randomly pick 1000, then pick from that.
-2. An ASR lookup is initiated and the ASR reading requires at least 50 calls. It does a sliding window over the last 7 days.
-3. If there are less than 50 calls to this CLI (or none), it will assume an ASR of 50%.
-4. It will pick the top 5 performing CLI's every 5 minutes.
+1. If there are more than 1000 records in the CLI Database, the system will randomly pick 1000 DIDs, then pick the best ones from that list of best performing ones.
+2. An ASR lookup is initiated and the ASR reading requires at least 50 calls (per CLI). It means the system will then shortlist 50 calls (out of 1000) with the best ASR.
+3. The system uses a **Sliding Window** concept over 7 days to achieve the ASR readings. This concept will add up the all the calls (successful and unsuccessful) per CLI to give us the ASR readings. Once the system gets all the CLIs ASR readings, it will arrange it from Best to Worse and will pick the top 5 CLIs.
```
Comment (AdamNawaz-connex) [inline docs/customer/cli.md:281]: no need for this

File: docs/customer/cli.md [hunk 19]
```diff
@@ -268,26 +268,32 @@ Our Userspace database allows you to manage large lists of numbers. Once you upl
...
+4. Out of the 1000 calls the system will pick the top 5 performing CLIs every 5 minutes.
+5. If there are less than 50 calls to this CLI (or none), it will assume an ASR of 50%.
```
Comment (AdamNawaz-connex) [inline docs/customer/cli.md:283]: Added above

File: docs/customer/cli.md [hunk 20]
```diff
@@ -268,26 +268,32 @@ Our Userspace database allows you to manage large lists of numbers. Once you upl
...
 **How does this algorithm work?**
 
-1. If there is more than 1000 records in the CLI Database, it will randomly pick 1000, then pick from that.
-2. An ASR lookup is initiated and the ASR reading requires at least 50 calls. It does a sliding window over the last 7 days.
-3. If there are less than 50 calls to this CLI (or none), it will assume an ASR of 50%.
-4. It will pick the top 5 performing CLI's every 5 minutes.
+1. If there are more than 1000 records in the CLI Database, the system will randomly pick 1000 DIDs, then pick the best ones from that list of best performing ones.
+2. An ASR lookup is initiated and the ASR reading requires at least 50 calls (per CLI). It means the system will then shortlist 50 calls (out of 1000) with the best ASR.
+3. The system uses a **Sliding Window** concept over 7 days to achieve the ASR readings. This concept will add up the all the calls (successful and unsuccessful) per CLI to give us the ASR readings. Once the system gets all the CLIs ASR readings, it will arrange it from Best to Worse and will pick the top 5 CLIs.
+4. Out of the 1000 calls the system will pick the top 5 performing CLIs every 5 minutes.
```
Comment (AdamNawaz-connex) [inline docs/customer/cli.md:282]: The number of top-performing CLIs selected is determined by the Performance Top Batch Size setting.
These top CLIs are selected for use based on the Performance Interval setting, which dictates how long this batch remains active.

File: docs/setup/appstore.md (hunk 1 above)
Comment (Bani2791) [reply docs/setup/appstore.md:7]: done

File: docs/setup/appstore.md (hunk 2 above)
Comment (Bani2791) [reply docs/setup/appstore.md:9]: done

File: docs/setup/appstore.md (hunk 3 above)
Comment (Bani2791) [reply docs/setup/appstore.md:18]: done

File: docs/anyedge/anyedge.md (hunk 4 above)
Comment (Bani2791) [reply docs/anyedge/anyedge.md:73]: done

File: docs/apps/architecture/query-builder.md (hunk 6 above)
Comment (Bani2791) [reply docs/apps/architecture/query-builder.md:40]: done

File: docs/circuit-test.md [hunk 21]
```diff
@@ -44,6 +44,49 @@ View a history of completed circuit tests, including the provider the call route
 + **Filters** pop out on the right allowing you to filter some of your data.
 + Adjust the Column ordering.
 
+|Parameter|Description|
+|---------|-----------|
+|**Created**| When the log was created mm/dd/yyyy|
```
[the remark this answers is not in the document]
Comment (Bani2791) [reply docs/circuit-test.md:49]: done

File: docs/circuit-test.md (hunk 7 above)
Comment (Bani2791) [reply docs/circuit-test.md:51]: done

File: docs/circuit-test.md (hunk 8 above)
Comment (Bani2791) [reply docs/circuit-test.md:52]: done

File: docs/circuit-test.md (hunk 9 above)
Comment (Bani2791) [reply docs/circuit-test.md:53]: done

File: docs/circuit-test.md [hunk 22]
```diff
@@ -44,6 +44,49 @@ View a history of completed circuit tests, including the provider the call route
...
+|**Provider**|List of your providers|
+|**MOS- Mean Opinion Score**|To find the call quality|
+|**DTMF- Dual Tone Multi Frequency**|diagnose and resolve various communication problems, enhancing network performance and ensuring smooth communication experiences|
+|**Answered**|Call has been connected and answered|
+|**Destroyed**|Call has been killed|
```
[the remark this answers is not in the document]
Comment (Bani2791) [reply docs/circuit-test.md:54]: done

File: docs/circuit-test.md (hunk 10 above)
Comment (Bani2791) [reply docs/circuit-test.md:55]: done

File: docs/circuit-test.md (hunk 11 above)
Comment (Bani2791) [reply docs/circuit-test.md:57]: done

File: docs/circuit-test.md (hunk 12 above)
Comment (Bani2791) [reply docs/circuit-test.md:61]: done

File: docs/circuit-test.md (hunk 13 above)
Comment (Bani2791) [reply docs/circuit-test.md:62]: done

File: docs/circuit-test.md (hunk 14 above)
Comment (Bani2791) [reply docs/circuit-test.md:78]: done

File: docs/customer/cdr.md (hunk 15 above)
Comment (Bani2791) [reply docs/customer/cdr.md:35]: done

File: docs/customer/cdr.md (hunk 17 above)
Comment (Bani2791) [reply docs/customer/cdr.md]: removed

File: docs/customer/cli.md [hunk 23]
```diff
@@ -268,26 +268,32 @@ Our Userspace database allows you to manage large lists of numbers. Once you upl
...
 
+This feature's objective is to allow customers to utilise the best performing CLIs.
+
 **How does this algorithm work?**
 
-1. If there is more than 1000 records in the CLI Database, it will randomly pick 1000, then pick from that.
-2. An ASR lookup is initiated and the ASR reading requires at least 50 calls. It does a sliding window over the last 7 days.
-3. If there are less than 50 calls to this CLI (or none), it will assume an ASR of 50%.
-4. It will pick the top 5 performing CLI's every 5 minutes.
+1. If there are more than 1000 records in the CLI Database, the system will randomly pick 1000 DIDs, then pick the best ones from that list of best performing ones.
```
[the remark this answers is not in the document]
Comment (Bani2791) [reply docs/customer/cli.md:279]: done

File: docs/customer/cli.md [hunk 24]
```diff
@@ -268,26 +268,32 @@ Our Userspace database allows you to manage large lists of numbers. Once you upl
...
+2. An ASR lookup is initiated and the ASR reading requires at least 50 calls (per CLI). It means the system will then shortlist 50 calls (out of 1000) with the best ASR.
```
[the remark this answers is not in the document]
Comment (Bani2791) [reply docs/customer/cli.md:280]: done

File: docs/customer/cli.md (hunk 18 above)
Comment (Bani2791) [reply docs/customer/cli.md:281]: done

File: docs/customer/cli.md (hunk 20 above)
Comment (Bani2791) [reply docs/customer/cli.md:282]: done

File: docs/customer/cli.md (hunk 19 above)
Comment (Bani2791) [reply docs/customer/cli.md:283]: done

File: docs/anyedge/anyedge.md (hunk 5 above)
Comment (Bani2791) [reply docs/anyedge/anyedge.md:76]: done

**Review by AdamNawaz-connex — APPROVED** [review]

**Pull request merged**